File: api/ai.js
export default async function handler(req, res) {
  // Handle CORS
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Method not allowed' });
  }

  try {
    const { token, prompt } = req.body;

    if (!token || !prompt) {
      return res.status(400).json({ error: 'Missing token or prompt' });
    }

    const response = await fetch(
      'https://router.huggingface.co/hf-inference/models/mistralai/Mixtral-8x7B-Instruct-v0.1',
      {
        method: 'POST',
        headers: {
          'Authorization': `Bearer ${token}`,
          'Content-Type': 'application/json'
        },
        body: JSON.stringify({
          inputs: prompt,
          parameters: {
            max_new_tokens: 500,
            temperature: 0.7,
            return_full_text: false
          }
        })
      }
    );

    if (!response.ok) {
      const errorText = await response.text();
      return res.status(response.status).json({ error: errorText });
    }

    const result = await response.json();
    const generatedText = result[0]?.generated_text || '';

    return res.status(200).json({ text: generatedText });

  } catch (error) {
    return res.status(500).json({ error: error.message });
  }
}

File: api/ai.py
from flask import Flask, request, jsonify
import requests

app = Flask(__name__)

@app.route('/api/ai', methods=['POST'])
def ai_proxy():
    """Proxy requests to Hugging Face API to avoid CORS issues"""
    try:
        data = request.get_json()
        token = data.get('token')
        prompt = data.get('prompt')

        if not token or not prompt:
            return jsonify({'error': 'Missing token or prompt'}), 400

        response = requests.post(
            'https://router.huggingface.co/hf-inference/models/mistralai/Mixtral-8x7B-Instruct-v0.1',
            headers={
                'Authorization': f'Bearer {token}',
                'Content-Type': 'application/json'
            },
            json={
                'inputs': prompt,
                'parameters': {
                    'max_new_tokens': 500,
                    'temperature': 0.7,
                    'return_full_text': False
                }
            },
            timeout=60
        )

        if not response.ok:
            return jsonify({'error': response.text}), response.status_code

        result = response.json()
        generated_text = result[0].get('generated_text', '') if result else ''

        return jsonify({'text': generated_text})

    except requests.Timeout:
        return jsonify({'error': 'Request timed out. The AI model may be loading, please try again.'}), 504
    except Exception as e:
        return jsonify({'error': str(e)}), 500
